credoai/modules/model_assessments/nlp_generator.py

```diff
@@ -0,0 +1,496 @@
+import matplotlib
+import matplotlib.pyplot as plt
+import pandas as pd
+import os
+import seaborn as sns
+
+from ._nlp_constants import PROMPTS_PATHS
+from absl import logging
+from credoai.data.utils import get_data_path
+from credoai.modules.credo_module import CredoModule
+from credoai.reporting import plot_utils
+from credoai.utils.common import wrap_list
+from datetime import datetime
+from googleapiclient import discovery
+from time import sleep
+
+matplotlib.rcParams.update({"font.size": 12})
+matplotlib.rc("figure", dpi=200)
+
+
+class NLPGeneratorAnalyzer(CredoModule):
+    """
+    NLP generation analyzer for Credo AI
+
+    This module assesses language generation models based on various prompts and assessment attributes
+
+    Parameters
+    ----------
+    generation_config : dict
+        generation models configuration dictionary where keys are names of the models and values are their callable generation functions (input : str, output: str)
+        Example: my_generation_config = {'gpt1': gpt1_text_generator, 'gpt2': gpt2_text_generator}
+
+    assessment_config : dict
```

Comment:
This whole `assessment_config` confuses me. These seem like they should be arguments for initializing the NLPGeneratorAnalyzer, and are somewhat needlessly buried in a separate dict, making documentation difficult.

Reply:
updated now to more clear design
```
        prompts,
        generation_functions,
        assessment_functions,
        perspective_config=None,
```